vocabularies.md

```diff
@@ -0,0 +1,283 @@
+# Controlled Vocabularies Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+All the endpoints here described are restricted to authenticated users.
+
+## Main Endpoint
+**/api/submission/vocabularies**   
+
+Provide access to the configured controlled vocabularies. It returns the list of existent controlled vocabularies.
+
+Example:
+```json
+{
+  "_embedded": {
+    "vocabularies": [
+      {
+        "id": "srsc",
+        "name": "srsc",
+        "scrollable": false,
+        "hierarchical": true,
+        "preloadLevel": 1,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc"
+          }
+        }
+      },
+      {
+        "id": "common_types",
+        "name": "common_types",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc"
+          }
+        }
+      },
+      {
+        "id": "common_iso_languages",
+        "name": "common_iso_languages",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/common_iso_languages/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/common_iso_languages"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 3,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+## Single Vocabulary
+**/api/submission/vocabularies/<:vocabulary-name>**
+
+Provide detailed information about a specific controlled vocabulary. The JSON response document is as follow
+```json
+{
+  "id": "srsc",
+  "name": "srsc",
+  "scrollable": false,
+  "hierarchical": true,
+  "preloadLevel": 1,
+  "type": "vocabulary"
+}
+```
+
+Attributes
```

Comment:
Wasn't it planned to also contain a boolean authority?

Reply:
it seems possible to avoid to introduce such attribute because the authority will be exposed only where it is liked to store it. In the srsc authority for instance where we need the "id" to allow the hierarchical browsing the authority is NOT included in the vocabularyEntry see
https://github.com/DSpace/Rest7Contract/pull/128/files#diff-1a0f138d702d2d84a2c28904e36e8150R215

https://github.com/DSpace/Rest7Contract/pull/128/files#diff-1a0f138d702d2d84a2c28904e36e8150R224

Reply:
For storage that would indeed work, but how can the submission UI determine when to display the lookup button if it's only determined in the returned values?

Reply:
this is the job of the submission-form definition endpoint. Looking to that I get the chance to update the attribute name to reflect the new wording (controlledVocabulary instead than authority)

Reply:
So you're suggesting the submission-form definition endpoint is extended to add an `"authority": true` to ensure the UI can identify whether the lookup button should be displayed?
<img width="597" alt="Schermafdruk 2020-06-25 12 08 49" src="https://user-images.githubusercontent.com/1528033/85700474-b42b0500-b6dc-11ea-87c1-170eaeaf2aeb.png">
Or will it always display the lookup unless the submission form type is dropdown?

Reply:
no, the UI should use the type and the controlledVocabulary attribute in the formField to determine which widget is needed (no real changes on this aspect compared to what is already implemented in the master).
It can be a dropdown, a lookup button or a suggest

Reply:
So the plan for determining the UI would be:
* If in the submission form type=dropdown => the vocabulary as a dropdown (and hierarchical is not supported)
* Otherwise, if in vocabulary definition the hierarchical=true => render as a hierarchical browse and suggest
* Otherwise if there's a vocabulary => render as a suggest
* Support for [lookup](https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L1507) is not exposed yet => needs to be handled as a separate ticket?

Reply:
I have added more details about the submission forms, no changes are proposed here is just to document what exists https://github.com/DSpace/Rest7Contract/pull/128/commits/f7076e53b9f3674eeb3a619976325c96f0b85567

Reply:
Thanks for documenting this functionality. Since this is new in 7.0 and was not documented yet, I wasn't aware of that implementation
I didn't verify whether it supports https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L1507 as well, or only adds the lookup buttons based on the submission-forms. But that's an implementation detail, and not a contract detail.

With this info, I assume the plan for determining the UI would be:
* If in the submission form type=dropdown => the vocabulary as a dropdown (and hierarchical is not supported)
* If in the submission form type=lookup or lookup-name => render as a lookup with detailed metadata similar to https://wiki.lyrasis.org/display/DSDOC6x/ORCID+Integration#ORCIDIntegration-SubmissionofnewDSpaceitems-Authorlookup
* Otherwise, if in vocabulary definition the hierarchical=true => render as a hierarchical browse and suggest
* Otherwise if there's a vocabulary => render as a suggest

Since this is complicated set of rules, I would recommend this is documented

Reply:
I agree that we need to provide advice to the UI / client on how this information should be used.  I've added suggestions inline in this Contract for how to document what @benbosman noted above. See this inline comment: https://github.com/DSpace/Rest7Contract/pull/128/files#r447073406